File: evidence/generated-systems/angular/inventory-on-angular/src/app/models/OutboundAllocation.js

// Define collection and schema for OutboundAllocation
export interface OutboundAllocation {
    allocationNumber:
	type : string
    allocatedQuantity:
	type : String
    allocationDate:
	type : Date
    Warehouse:
	type : Schema.Types.ObjectId
    Sku:
	type : Schema.Types.ObjectId
    InventoryItem:
	type : Schema.Types.ObjectId
    Reservation:
	type : Schema.Types.ObjectId
    Lot:
	type : Schema.Types.ObjectId
    SerialNumbers:
 	type : [{ type: Schema.Types.ObjectId, ref: 'SerialNumber' }]
    SourceLocation:
	type : Schema.Types.ObjectId
    Status:
 	type : String
#
    collection: 'outboundAllocations'
}

File: evidence/generated-systems/golang/inventory-on-golang/inventory-on-golang/internal/model/OutboundAllocation.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// OutboundAllocation Declaration
//==============================================================
type OutboundAllocation struct {
    gorm.Model
     AllocationNumber                                    string
    AllocatedQuantity                                                            string
    AllocationDate                                                            time.Time
    WarehouseId         *uint
    Warehouse           *Warehouse `gorm:"foreignKey:WarehouseId"`
    SkuId         *uint
    Sku           *StockKeepingUnit `gorm:"foreignKey:SkuId"`
    InventoryItemId         *uint
    InventoryItem           *InventoryItem `gorm:"foreignKey:InventoryItemId"`
    ReservationId         *uint
    Reservation           *Reservation `gorm:"foreignKey:ReservationId"`
    LotId         *uint
    Lot           *Lot `gorm:"foreignKey:LotId"`
     SerialNumbers           []SerialNumber `gorm:"foreignKey:SerialNumbersFromOutboundAllocationId"`
    SourceLocationId         *uint
    SourceLocation           *StorageLocation `gorm:"foreignKey:SourceLocationId"`
    Status                      AllocationStatus

// parent associations as their child

}

